Sources/TokamakGTK/GSignal.swift

```diff
@@ -100,3 +146,9 @@ final class SingleParamClosureBox<T, U> {
 
   init(_ closure: @escaping (T) -> U) { self.closure = closure }
 }
+
+final class DualParamClosureBox<T, U, V> {
+  let closure: (T, U) -> V
+
+  init(_ closure: @escaping (T, U) -> V) { self.closure = closure }
```

Comment:
It's a shame that Swift doesn't support variadic generics, or macros, or something of that nature that could simplify this boilerplate. I guess something like GYB than it's worth, but we could try it later you want 🙂

Reply:
Right, there are quite a few good use cases for variadic generics. I hope that we'll see them one day.
I haven't looked too much into the various events/signals in gtk, but my guess is that they don't have callbacks with a whole lot of arguments... So until we see some more of them I guess this is acceptable.